### ISSUE TITLE
KOLIBRI SCRIPT DIR not set

### ISSUE THREAD

**Issue by radinamatic** [learningequality/kolibri-installer-windows#18]
Fresh Windows 7 VM snapshot, installed Kolibri build from [#1816](https://github.com/learningequality/kolibri/pull/1816), got these env vars:

```
Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation.  All rights reserved.

C:\Users\IEUser>set
ALLUSERSPROFILE=C:\ProgramData
APPDATA=C:\Users\IEUser\AppData\Roaming
CommonProgramFiles=C:\Program Files\Common Files
COMPUTERNAME=IE11WIN7
ComSpec=C:\Windows\system32\cmd.exe
FP_NO_HOST_CHECK=NO
HOMEDRIVE=C:
HOMEPATH=\Users\IEUser
KOLIBRI_SCRIPT_DIR=C:\Python27\Scripts
LOCALAPPDATA=C:\Users\IEUser\AppData\Local
LOGONSERVER=\\IE11WIN7
NUMBER_OF_PROCESSORS=1
OS=Windows_NT
Path=C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32
\WindowsPowerShell\v1.0\
PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC
PROCESSOR_ARCHITECTURE=x86
PROCESSOR_IDENTIFIER=x86 Family 6 Model 55 Stepping 8, GenuineIntel
PROCESSOR_LEVEL=6
PROCESSOR_REVISION=3708
ProgramData=C:\ProgramData
ProgramFiles=C:\Program Files
PROMPT=$P$G
PSModulePath=C:\Windows\system32\WindowsPowerShell\v1.0\Modules\
PUBLIC=C:\Users\Public
SESSIONNAME=Console
SystemDrive=C:
SystemRoot=C:\Windows
TEMP=C:\Users\IEUser\AppData\Local\Temp
TMP=C:\Users\IEUser\AppData\Local\Temp
USERDOMAIN=IE11WIN7
USERNAME=IEUser
USERPROFILE=C:\Users\IEUser
windir=C:\Windows

C:\Users\IEUser>
```

The same install on Win 10 VM:

```
Microsoft Windows [Version 10.0.15063]
(c) 2017 Microsoft Corporation. All rights reserved.

C:\Users\IEUser>set
ALLUSERSPROFILE=C:\ProgramData
APPDATA=C:\Users\IEUser\AppData\Roaming
CommonProgramFiles=C:\Program Files\Common Files
CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files
CommonProgramW6432=C:\Program Files\Common Files
COMPUTERNAME=MSEDGEWIN10
ComSpec=C:\Windows\system32\cmd.exe
CYGWIN=mintty
HOMEDRIVE=C:
HOMEPATH=\Users\IEUser
KOLIBRI_SCRIPT_DIR=C:\Python27\Scripts
LOCALAPPDATA=C:\Users\IEUser\AppData\Local
LOGONSERVER=\\MSEDGEWIN10
NUMBER_OF_PROCESSORS=1
OneDrive=C:\Users\IEUser\OneDrive
OS=Windows_NT
Path=C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files\OpenSSH\bin;C:\Users\IEUser\AppData\Local\Microsoft\WindowsApps;
PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC
PROCESSOR_ARCHITECTURE=AMD64
PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 55 Stepping 8, GenuineIntel
PROCESSOR_LEVEL=6
PROCESSOR_REVISION=3708
ProgramData=C:\ProgramData
ProgramFiles=C:\Program Files
ProgramFiles(x86)=C:\Program Files (x86)
ProgramW6432=C:\Program Files
PROMPT=$P$G
PSModulePath=C:\Program Files\WindowsPowerShell\Modules;C:\Windows\system32\WindowsPowerShell\v1.0\Modules
PUBLIC=C:\Users\Public
SESSIONNAME=Console
SystemDrive=C:
SystemRoot=C:\Windows
TEMP=C:\Users\IEUser\AppData\Local\Temp
TMP=C:\Users\IEUser\AppData\Local\Temp
USERDOMAIN=MSEDGEWIN10
USERDOMAIN_ROAMINGPROFILE=MSEDGEWIN10
USERNAME=IEUser
USERPROFILE=C:\Users\IEUser
windir=C:\Windows

C:\Users\IEUser>
```

So far so good. 

However, if there was a KA Lite installed previously, (@laurenlichtman had this problem), `KALITE_PYTHON=\Python27\python.exe` remained in the registry even after KA Lite uninstall, and for some reason that might have interfered with properly setting up the `KOLIBRI_SCRIPT_DIR`. This resulted in the infamous error notification:

![img_20170626_195125_1024](https://user-images.githubusercontent.com/1457929/28082536-054d23ae-6673-11e7-89f9-1f53534d140c.jpg)
   
With this error it seems impossible to start Kolibri from the taskbar icon, but it does start from the command prompt. As you notice in her env vars, `KALITE_PYTHON` is there, but no `KOLIBRI_SCRIPT_DIR`: 

```
C:\Users\ljl21>set 
ALLUSERSPROFILE=C:\ProgramData
APPDATA=C:\Users\ljl21\AppData\Roaming
CommonProgramFiles=C:\Program Files\Common Files
CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files
CommonProgramW6432=C:\Program Files\Common Files
COMPUTERNAME=DESKTOP-09E4QED
ComSpec=C:\windows\system32\cmd.exe
HOMEDRIVE=C:
HOMEPATH=\Users\ljl21
KALITE_PYTHON=\Python27\python.exe
LOCALAPPDATA=C:\Users\ljl21\AppData\Local
LOGONSERVER=\\DESKTOP-09E4QED
NUMBER_OF_PROCESSORS=4
OneDrive=C:\Users\ljl21\OneDrive
OS=Windows_NT
Path=C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Skype\Phone\;C:\Users\ljl21\AppData\Local\Microsoft\WindowsApps;
PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC
PROCESSOR_ARCHITECTURE=AMD64
PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 78 Stepping 3, GenuineIntel
PROCESSOR_LEVEL=6
PROCESSOR_REVISION=4e03
ProgramData=C:\ProgramData
ProgramFiles=C:\Program Files
ProgramFiles(x86)=C:\Program Files (x86)
ProgramW6432=C:\Program Files
PROMPT=$P$G
PSModulePath=C:\Program Files\WindowsPowerShell\Modules;C:\windows\system32\WindowsPowerShell\v1.0\Modules
PUBLIC=C:\Users\Public
SESSIONNAME=Console
SystemDrive=C:
SystemRoot=C:\windows
TEMP=C:\Users\ljl21\AppData\Local\Temp
TMP=C:\Users\ljl21\AppData\Local\Temp
USERDOMAIN=DESKTOP-09E4QED
USERDOMAIN_ROAMINGPROFILE=DESKTOP-09E4QED
USERNAME=ljl21
USERPROFILE=C:\Users\ljl21
windir=C:\windows
```

Another partner had the same issue, it was impossible to start Kolibri from the taskbar icon, but it does start from the command prompt. However in this case there was no `KALITE_PYTHON` var, but no `KOLIBRI_SCRIPT_DIR` either: 

```
Microsoft Windows [Version 6.3.9600]
(c) 2013 Microsoft Corporation. All rights reserved.

C:\Users\jakao>set
AGORA_HOST=agora-offline
AGORA_INSTALL=C:\Agora\OfflinePlayer\us
AGORA_INSTALL_WWW=C:\Agora\OfflinePlayer\us\www
ALLUSERSPROFILE=C:\ProgramData
APPDATA=C:\Users\jakao\AppData\Roaming
CommonProgramFiles=C:\Program Files\Common Files
CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files
CommonProgramW6432=C:\Program Files\Common Files
COMPUTERNAME=994380APLW1237
ComSpec=C:\windows\system32\cmd.exe
FP_NO_HOST_CHECK=NO
HOMEDRIVE=C:
HOMEPATH=\Users\jakao
LOCALAPPDATA=C:\Users\jakao\AppData\Local
LOGONSERVER=\\994380APLW1237
NUMBER_OF_PROCESSORS=4
OS=Windows_NT
Path=C:\ProgramData\Oracle\Java\javapath;C:\windows\system32;C:\windows;C:\windo
ws\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x
86)\Skype\Phone\;C:\Program Files (x86)\QuickTime\QTSystem\;C:\Program Files (x8
6)\Intel\OpenCL SDK\2.0\bin\x86;C:\Program Files (x86)\Intel\OpenCL SDK\2.0\bin\
x64
PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC
PHP_INI_SELECT=php_production.ini
PHP_SELECT=php54
PROCESSOR_ARCHITECTURE=AMD64
PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 42 Stepping 7, GenuineIntel
PROCESSOR_LEVEL=6
PROCESSOR_REVISION=2a07
ProgramData=C:\ProgramData
ProgramFiles=C:\Program Files
ProgramFiles(x86)=C:\Program Files (x86)
ProgramW6432=C:\Program Files
PROMPT=$P$G
PSModulePath=C:\windows\system32\WindowsPowerShell\v1.0\Modules\
PUBLIC=C:\Users\Public
SESSIONNAME=Console
SystemDrive=C:
SystemRoot=C:\windows
TEMP=C:\Users\jakao\AppData\Local\Temp
TMP=C:\Users\jakao\AppData\Local\Temp
USERDOMAIN=994380APLW1237
USERDOMAIN_ROAMINGPROFILE=994380APLW1237
USERNAME=jakao
USERPROFILE=C:\Users\jakao
windir=C:\windows

C:\Users\jakao>
```

So I see two potential issues here for which we need solutions:

1. How to avoid interference from env vars remaining form previous KA Lite installs to Kolibri install, as there will potentially be many users upgrading from KA Lite to Kolibri.

2. Investigate what else can interfere the proper setting up of the `KOLIBRI_SCRIPT_DIR`.

**Comment by laurenlichtman** [comment]
On Windows 10, I was running 0.4.0.dev448 (Chrome) and I pressed "Stop server" in the tray and then the same error message appeared (env var `kolibri_script_dir` is not set). I checked the environment variables and it created a variable for my user with a blank value. Kolibri still appears to be working from the browser but I'm unable to "stop server" from the system tray.

**Comment by aronasorman** [comment]
@mrpau-eduard any clue on what the issue is?

**Comment by mrpau-richard** [comment]
@radinamatic & @aronasorman I think the issue is the existing kolibri static installed when installing the kolibri installer with the same version. here's my steps to reproduce this issue.

- Install kolibri.
- Uninstall kolibri.
- Install kolibri again with the same version.
- The `KOLIBRI_SCRIPT_DIR` is not present at user environment variables.

To fix this issue manually set the `KOLIBRI_SCRIPT_DIR` environment variable to `C:\Python27\Scripts` or manually uninstall the kolibri static before installing kolibri.

I have existing [PR](https://github.com/learningequality/kolibri-installer-windows/pull/16/files) to fix this issue.

**Comment by radinamatic** [comment]
Confirmed, this should be fixed, as soon as the above PR by @mrpau-richard is merged! :+1:

**Comment by mrpau-richard** [comment]
Fixed in #20, closing this.
